### PR TITLE
unit tests for processors

### PR DESCRIPTION
**Pull request by tnawathe21**
## Description
This PR modifies the existing header rewrite unit tests to reflect recent changes in the `Processor` interfaces. It also adds unit tests for `ConditionProcessor`.

## Testing
```
$ bazel test //header-rewrite-filter:header_processor_test
INFO: Analyzed target //header-rewrite-filter:header_processor_test (0 packages loaded, 0 targets configured).
INFO: Found 1 test target...
Target //header-rewrite-filter:header_processor_test up-to-date:
  bazel-bin/header-rewrite-filter/header_processor_test
INFO: Elapsed time: 0.402s, Critical Path: 0.05s
INFO: 1 process: 1 internal.
INFO: Build completed successfully, 1 total action

Executed 0 out of 1 test: 1 test passes.
```

### CODE REVIEW

**Comment by tnawathe21** [comment]
Current dependencies on/for this PR:
* main
  * **PR #30** <a href="https://app.graphite.dev/github/pr/DataDog/envoy-filter-example/30" target="_blank">Graphite</a>  👈
    * **PR #31** <a href="https://app.graphite.dev/github/pr/DataDog/envoy-filter-example/31" target="_blank">Graphite</a> 
    * **PR #25** <a href="https://app.graphite.dev/github/pr/DataDog/envoy-filter-example/25" target="_blank">Graphite</a> 
      * **PR #26** <a href="https://app.graphite.dev/github/pr/DataDog/envoy-filter-example/26" target="_blank">Graphite</a> 
        * **PR #33** <a href="https://app.graphite.dev/github/pr/DataDog/envoy-filter-example/33" target="_blank">Graphite</a> 
          * **PR #34** <a href="https://app.graphite.dev/github/pr/DataDog/envoy-filter-example/34" target="_blank">Graphite</a> 

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/DataDog/envoy-filter-example/30?utm_source=stack-comment).

File: header-rewrite-filter/header_processor_test.cc [hunk 1]
```diff
@@ -14,85 +14,327 @@ class ProcessorTest : public ::testing::Test {
...
     EXPECT_TRUE(status == absl::OkStatus());
-    set_header_processor.executeOperation(headers);
-    EXPECT_EQ("mock_val1", headers.get(Http::LowerCaseString("mock_key"))[0]->value().getStringView());
-    EXPECT_EQ("mock_val2", headers.get(Http::LowerCaseString("mock_key"))[1]->value().getStringView());
+    EXPECT_EQ("another_mock_value", headers.get(Http::LowerCaseString("mock_header"))[0]->value().getStringView());
+
+    // can't set multiple values
+    operation_expression = {"http-response", "set-header", "mock_key", "mock_val1", "mock_val2"};
+    status = set_header_processor.parseOperation(operation_expression, (operation_expression.begin() + 2));
+    EXPECT_TRUE(status != absl::OkStatus());
```
Comment (henrymwang) [inline header-rewrite-filter/header_processor_test.cc]: Could you check the expected status instead?

File: header-rewrite-filter/header_processor_test.cc [hunk 2]
```diff
@@ -14,85 +14,327 @@ class ProcessorTest : public ::testing::Test {
...
     operation_expression = {"http-response", "set-header", "mock_key"};
-    status = set_header_processor.parseOperation(operation_expression);
+    status = set_header_processor.parseOperation(operation_expression, (operation_expression.begin() + 2));
     EXPECT_TRUE(status != absl::OkStatus());
     EXPECT_EQ(status.message(), "not enough arguments for set-header");
 
+    // empty condition
+    operation_expression = {"http-response", "set-header", "mock_key", "mock_val", "if"};
+    status = set_header_processor.parseOperation(operation_expression, (operation_expression.begin() + 2));
+    EXPECT_TRUE(status != absl::OkStatus());
```
Comment (henrymwang) [inline header-rewrite-filter/header_processor_test.cc]: same

File: header-rewrite-filter/header_processor_test.cc [hunk 3]
```diff
@@ -14,85 +14,327 @@ class ProcessorTest : public ::testing::Test {
...
+    EXPECT_TRUE(status == absl::OkStatus());
+    status = set_header_processor.executeOperation(headers, nullptr);
     EXPECT_TRUE(status == absl::OkStatus());
-    set_header_processor.executeOperation(headers);
     EXPECT_EQ("mock_value", headers.get(Http::LowerCaseString("mock_header"))[0]->value().getStringView());
 
-    // can set multiple values
-    operation_expression = {"http-response", "set-header", "mock_key", "mock_val1", "mock_val2"};
-    status = set_header_processor.parseOperation(operation_expression);
+    // replaces already existing value
```
Comment (henrymwang) [inline header-rewrite-filter/header_processor_test.cc]: It seems like a lot of this code is being repeated and you're also mutating the same `headers` object multiple times which could make it tricky to debug issues in the future. Can we instead create a vector of `std::vector<absl::string_view>` as the test cases and for each test case just create a new `SetHeaderProcessor` / `TestRequestHeaderMapImpl`, run the processor's parse and execute, and check the result?

https://github.com/envoyproxy/envoy/blob/cacd32bc9e3285f6eee7efbc3fcb3e7291975055/test/common/http/path_utility_test.cc#L74 is an example

**Review by henrymwang — APPROVED** [review]

**Review by henrymwang — APPROVED** [review]

**Pull request merged**